src/MSBuild/XMake.cs

```diff
@@ -1259,10 +1259,12 @@ private static BuildResult ExecuteBuild(BuildManager buildManager, BuildRequestD
             lock (s_buildLock)
             {
                 var projectGraph = new ProjectGraph(projectFile, projectCollection, globalProperties, toolsVersion);
+                var targetLists = projectGraph.GetTargetLists(targets);
 
                 // TODO: Do a full graph traversal
                 var projectInstance = projectGraph.EntryProjectNode.Project.CreateProjectInstance();
-                var request = new BuildRequestData(projectInstance, targets);
+                var targetList = targetLists[projectGraph.EntryProjectNode];
```

Comment:
Yes, I know `targetList == targets`. This ensures that it actually works though, and once we implement the full graph traversal for build, it'll look more like this anyway.